triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala

```diff
@@ -191,17 +191,21 @@ object Server {
       failureRetryTimeRange: Duration,
       dar: Option[Dar[(PackageId, Package)]],
       jdbcConfig: Option[JdbcConfig],
+      noSecretKey: Boolean,
   ): Behavior[Message] = Behaviors.setup { ctx =>
     val triggerDao = jdbcConfig.map(TriggerDao(_)(ctx.system.executionContext))
 
     val key =
       sys.env.get("TRIGGER_SERVICE_SECRET_KEY") match {
         case Some(key) => key
         case None => {
-          val logMsg =
-            "WARNING : The environment variable 'TRIGGER_SERVICE_SECRET_KEY' is not defined. It is highly recommended that a non-empty value for this variable be set. If the service startup parameters do not include the '--no-secret-key' option, the service will now terminate."
-          ctx.log.info(logMsg)
-          "secret key"
+          ctx.log.warn(
+            "The environment variable 'TRIGGER_SERVICE_SECRET_KEY' is not defined. It is highly recommended that a non-empty value for this variable be set. If the service startup parameters do not include the '--no-secret-key' option, the service will now terminate.")
```

Comment:
I would give slightly different warnings in each of the cases below, rather than saying "if you haven't used 'no-secret-key'", since you have that information on hand.